k2/python/k2/fsa.py

```diff
@@ -70,23 +117,78 @@ def __setattr__(self, name: str, value: Any) -> None:
             object.__setattr__(self, name, value)
         elif isinstance(value, torch.Tensor):
             assert value.shape[0] == self.arcs.values().shape[0]
-            # TODO(fangjun): do we really need a copy here?
+            if name == 'labels':
+                assert value.dtype == torch.int32
+                self.arcs.values()[:, 2] = value
+                return
+
             self._tensor_attr[name] = value.clone()
```

Comment:
I don't see why we need to do clone() here.

Reply:
My intention was to retain a copy inside FSA so that its attribute is not modified accidentally.

Reply:
I'd rather just warn the user that it's not cloned so the user should be careful about modifying it.
In PyTorch in-place operations are fairly rare anyway.

Reply:
A caution is added and `clone` is removed.